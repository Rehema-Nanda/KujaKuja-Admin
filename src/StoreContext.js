// https://mobx-react.js.org/recipes-context

import React from "react";
import PropTypes from "prop-types";
import RootStore from "./stores/RootStore";

export const StoreContext = React.createContext(null);

export const StoreProvider = ({ children }) => {
    const store = new RootStore();
    return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};
StoreProvider.propTypes = {
    children: PropTypes.node.isRequired,
};

export const StoreConsumer = StoreContext.Consumer;

export const useStore = () => {
    const store = React.useContext(StoreContext);
    if (!store) {
        // this is especially useful in TypeScript so you don't need to be checking for null all the time
        throw new Error("useStore must be used within a StoreProvider.");
    }
    return store;
};
