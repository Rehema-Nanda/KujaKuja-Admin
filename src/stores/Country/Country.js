import {
    decorate, computed, observable, comparer,
} from "mobx";
import lineToPolygon from "@turf/line-to-polygon";

class Country {
    store = null;
    id = null;
    enabled = null;
    name = null;
    isoTwoLetterCode = null;
    geojson = null;
    lat = null;
    lng = null;
    createdAt = null;
    updatedAt = null;

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.id = json.id;
            this.updateFromJson(json);
        }
    }

    get asJson() {
        return {
            enabled: this.enabled,
            name: this.name,
            iso_two_letter_code: this.isoTwoLetterCode,
            geojson: JSON.stringify(this.geojson),
            lat: this.lat,
            lng: this.lng,
            created_at: this.createdAt,
            updated_at: this.updatedAt,
        };
    }

    updateFromJson(json) {
        this.enabled = json.enabled;
        this.name = json.name;
        this.isoTwoLetterCode = json.iso_two_letter_code;
        this.geojson = json.geojson;
        this.lat = json.lat;
        this.lng = json.lng;
        this.createdAt = json.created_at;
        this.updatedAt = json.updated_at;

        this.fixGeoJson();
    }

    fixGeoJson() {
        // TODO: https://trello.com/c/wxv8WY0m/426-frontend-and-admin-backend-are-both-doing-linestring-to-polygon-conversion-but-differently-json-parsing
        if (this.geojson && this.geojson.geometry.type === "LineString") {
            this.geojson = lineToPolygon(this.geojson);
        }
    }

    async delete() {
        if (!this.id) {
            return;
        }
        await this.store.apiService.deleteCountry(this.id);
        await this.store.removeCountry(this);
    }

    async save() {
        const json = this.asJson;
        if (this.id) {
            await this.store.apiService.updateCountry(this.id, json);
        }
        else {
            const responseData = await this.store.apiService.createCountry(json);
            this.id = responseData.id;
            await this.store.addCountry(this);
        }
    }
}
decorate(Country, {
    lat: observable,
    lng: observable,
    asJson: computed({ equals: comparer.shallow }),
});

export default Country;
