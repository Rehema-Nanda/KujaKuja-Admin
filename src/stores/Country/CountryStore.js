import {
    flow, decorate, observable, computed,
} from "mobx";
import Country from "./Country";
import CountryApiService from "./CountryApiService";
import AppConfig from "../../AppConfig";

class CountryStore {
    countries = [];
    count = 0;
    enabledCountries = [];
    isLoading = false;

    constructor(rootStore) {
        this.rootStore = rootStore;
        this.apiService = new CountryApiService(AppConfig.API_URL);

        this.loadCountries();
        this.loadEnabledCountries();
    }

    get countriesSortedByName() {
        return this.countries.slice().sort((a, b) => {
            if (a.name < b.name) {
                return -1;
            }
            else if (a.name > b.name) {
                return 1;
            }
            return 0;
        });
    }

    loadCountries = flow(function* (config) { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const countriesJson = yield this.apiService.fetchAllCountries(config);
            this.countries.replace(countriesJson.data.map((countryJson) => {
                return new Country(this, countryJson);
            }));
            this.count = countriesJson.count;
            this.isLoading = false;
        }
        catch (error) {
            console.error(`CountryStore failed to load countries. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    loadEnabledCountries = flow(function* () { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const countriesJson = yield this.apiService.fetchEnabledCountries();
            this.enabledCountries.replace(countriesJson.map((countryJson) => {
                return new Country(this, countryJson);
            }));
            this.isLoading = false;
        }
        catch (error) {
            console.error(`CountryStore failed to load enabled countries. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    updateCountryFromServer = flow(function* (id) { // eslint-disable-line func-names
        this.isLoading = true;
        const countriesJson = yield this.apiService.fetchCountryById(id);
        let country = this.countries.find((c) => {
            return c.id === id;
        });

        if (countriesJson.length === 0 && country) {
            // country no longer exists on the server, remove it locally
            yield this.removeCountry(country);
            country = null;
        }
        if (countriesJson.length > 0) {
            if (country) {
                country.updateFromJson(countriesJson[0]);
            }
            else {
                country = new Country(this, countriesJson[0]);
                this.countries.push(country);
            }
        }
        this.isLoading = false;
        return country;
    });

    removeCountry = flow(function* (country) { // eslint-disable-line func-names, require-yield
        this.countries.splice(this.countries.indexOf(country), 1);
    });

    addCountry = flow(function* (country) { // eslint-disable-line func-names, require-yield
        this.countries.push(country);
    });
}
decorate(CountryStore, {
    countries: observable,
    enabledCountries: observable,
    countriesSortedByName: computed,
    isLoading: observable,
});

export default CountryStore;
