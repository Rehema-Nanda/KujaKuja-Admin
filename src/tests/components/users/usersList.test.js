import React from "react";
import { mount } from "enzyme";
import UsersList from "../../../components/users/list";
import { StoreProvider } from "../../../StoreContext";
import CountryStore from "../../../stores/Country/CountryStore";
import Country from "../../../stores/Country/Country";
import LocationStore from "../../../stores/Location/LocationStore";
import Location from "../../../stores/Location/Location";
import UserStore from "../../../stores/User/UserStore";
import User from "../../../stores/User/User";

jest.mock("bcryptjs");
jest.mock("../../../stores/Country/CountryStore");
jest.mock("../../../stores/Location/LocationStore");
jest.mock("../../../stores/User/UserStore");

CountryStore.mockImplementation(() => {
    return {
        countries: [
            new Country(
                null,
                {
                    id: "1",
                    enabled: false,
                    name: "Afghanistan",
                    iso_two_letter_code: "AF",
                    geojson: null,
                    lat: "0.00000000",
                    lng: "0.00000000",
                    created_at: null,
                    updated_at: null,
                },
            ),
        ],
    };
});

LocationStore.mockImplementation(() => {
    return {
        locations: [
            new Location(
                null,
                {
                    id: "1",
                    name: "Test Location",
                    geojson: null,
                    lat: "0.00000000",
                    lng: "0.00000000",
                    created_at: null,
                    updated_at: null,
                    country_id: "1",
                    country_name: "Afghanistan",
                },
            ),
        ],
    };
});

UserStore.mockImplementation(() => {
    return {
        users: [
            new User(
                null,
                {
                    id: "1",
                    email: "testuser@kujakuja.com",
                    encrypted_password: "password",
                    current_sign_in_at: "2020-01-01T00:00:00.000Z",
                    created_at: null,
                    updated_at: null,
                    settlement_id: "1",
                    settlement_name: "Test Location",
                    is_admin: true,
                    is_survey: false,
                    is_service_provider: false,
                },
            ),
            new User(
                null,
                {
                    id: "2",
                    email: "testuser2@kujakuja.com",
                    encrypted_password: "password",
                    current_sign_in_at: "2020-01-01T00:00:00.000Z",
                    created_at: null,
                    updated_at: null,
                    settlement_id: "1",
                    settlement_name: "Test Location",
                    is_admin: true,
                    is_survey: false,
                    is_service_provider: false,
                },
            ),
        ],
        userInfo: [
            new User(
                null,
                {
                    id: "1",
                    email: "testuser@kujakuja.com",
                    encrypted_password: "password",
                    current_sign_in_at: "2020-01-01T00:00:00.000Z",
                    created_at: null,
                    updated_at: null,
                    settlement_id: "1",
                    settlement_name: "Test Location",
                    is_admin: true,
                    is_survey: false,
                    is_service_provider: false,
                },
            ),
        ],
        count: 2,
    };
});

describe("UsersList component", () => {
    const testProps = {
        history: {
            push: jest.fn(),
        },
    };
    const wrapper = mount(<UsersList {...testProps} />, {
        wrappingComponent: StoreProvider,
    });

    it("should render correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should display users table", () => {
        expect(wrapper.find("table")).toHaveLength(1);
        expect(wrapper.find("th").at(3).text()).toBe("Results: 2");
    });

    it("should push history when edit button is clicked", () => {
        wrapper.find("button").at(1).simulate("click");
        expect(testProps.history.push).toHaveBeenCalledWith("/users/1");
    });

    it("should push history when add button is clicked", () => {
        wrapper.find("button").first().simulate("click");
        expect(testProps.history.push).toHaveBeenCalledWith("/users/add");
    });

    it("should display a zero count on the users table", () => {
        UserStore.mockImplementationOnce(() => {
            return {
                users: [],
                count: 0,
            };
        });

        const wrapper2 = mount(<UsersList {...testProps} />, {
            wrappingComponent: StoreProvider,
        });
        expect(wrapper2.find("th").at(3).text()).toBe("Results: 0");
    });
});
