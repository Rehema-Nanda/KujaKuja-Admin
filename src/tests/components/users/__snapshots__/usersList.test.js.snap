// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList component should render correctly 1`] = `
"<UsersList history={{...}}>
  <Toolkitprovider keyField=\\"id\\" data={{...}} columns={{...}} bootstrap4={true}>
    <ToolkitProvider keyField=\\"id\\" data={{...}} columns={{...}} bootstrap4={true} search={false} exportCSV={false}>
      <div className=\\"page\\">
        <div className=\\"title-background-parent\\">
          <div className=\\"title-background-shape\\" />
        </div>
        <Banner pageTitle=\\"team members\\" elementAction=\\"list\\" buttonFunction={[Function]} buttonDisabled={false}>
          <Container tag=\\"div\\">
            <div className=\\"container\\">
              <Row tag=\\"div\\">
                <div className=\\"row\\">
                  <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
                    <div className=\\"col-sm-7 col-md-8\\">
                      <div className=\\"page-title\\">
                        <h1 className=\\"left\\">
                          Team members
                        </h1>
                        <p>
                          Manage your
                           
                          team members
                        </p>
                      </div>
                    </div>
                  </Col>
                  <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
                    <div className=\\"col-sm-5 col-md-4\\">
                      <div className=\\"page-title\\">
                        <Button disabled={false} color=\\"primary\\" className=\\"primary\\" onClick={[Function]} tag=\\"button\\">
                          <button type=\\"button\\" disabled={false} onClick={[Function: bound onClick]} className=\\"primary btn btn-primary\\" aria-label={{...}}>
                            Add Team members
                          </button>
                        </Button>
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </Container>
        </Banner>
        <Container tag=\\"div\\">
          <div className=\\"container\\">
            <div className=\\"white-page-bg\\">
              <Row tag=\\"div\\">
                <div className=\\"row\\">
                  <Col tag=\\"div\\" widths={{...}}>
                    <div className=\\"col\\">
                      <CustomizableProvider pagination={{...}}>
                        <StateProvider pagination={{...}}>
                          <div>
                            <BootstrapTableContainer keyField=\\"id\\" columns={{...}} data={{...}} bootstrap4={true} setDependencyModules={[Function: bound setDependencyModules]} registerExposedAPI={[Function]} pagination={{...}} setPaginationRemoteEmitter={[Function]} dataChangeListener={{...}} remote={true} defaultSorted={{...}} striped={true} hover={true} bordered={false} noDataIndication=\\"No Data\\" onTableChange={[Function]}>
                              <DataProvider keyField=\\"id\\" columns={{...}} data={{...}}>
                                <SortProvider keyField=\\"id\\" columns={{...}} defaultSorted={{...}} defaultSortDirection={[undefined]} sort={[undefined]} data={{...}}>
                                  <PaginationDataProvider pagination={{...}} data={{...}} bootstrap4={true} isRemotePagination={[Function]} remoteEmitter={{...}} onDataSizeChange={[undefined]}>
                                    <ColumnManagementProvider keyField=\\"id\\" columns={{...}} toggles={{...}}>
                                      <BootstrapTable keyField=\\"id\\" columns={{...}} data={{...}} bootstrap4={true} setDependencyModules={[Function: bound setDependencyModules]} registerExposedAPI={[Function]} pagination={{...}} setPaginationRemoteEmitter={[Function]} dataChangeListener={{...}} remote={true} defaultSorted={{...}} striped={true} hover={true} bordered={false} noDataIndication=\\"No Data\\" onTableChange={[Function]} sortOrder=\\"asc\\" onSort={[Function]} sortField=\\"email\\" setRemoteEmitter={[undefined]} condensed={false} selectRow={{...}} expandRow={{...}} cellEdit={{...}} filterPosition=\\"inline\\">
                                        <div className=\\"react-bootstrap-table\\">
                                          <table id={[undefined]} className=\\"table table-striped table-hover\\">
                                            <Header columns={{...}} className={[undefined]} wrapperClasses={[undefined]} sortField=\\"email\\" sortOrder=\\"asc\\" onSort={[Function]} globalSortCaret={[undefined]} onFilter={[undefined]} currFilters={[undefined]} onExternalFilter={[undefined]} selectRow={{...}} expandRow={{...}} filterPosition=\\"inline\\">
                                              <thead className={[undefined]}>
                                                <tr className={[undefined]}>
                                                  <HeaderCell index={0} column={{...}} onSort={[Function]} sorting={true} sortOrder=\\"asc\\" globalSortCaret={[undefined]} isLastSorting={true} onFilter={[undefined]} currFilters={[undefined]} onExternalFilter={[undefined]} filterPosition=\\"inline\\">
                                                    <th tabIndex={0} aria-label=\\"Email sort asc\\" onKeyUp={[Function]} onClick={[Function]} className=\\"sortable\\">
                                                      Email
                                                      <SortCaret order=\\"asc\\">
                                                        <span className=\\"caret-4-asc\\" />
                                                      </SortCaret>
                                                    </th>
                                                  </HeaderCell>
                                                  <HeaderCell index={1} column={{...}} onSort={[Function]} sorting={false} sortOrder=\\"asc\\" globalSortCaret={[undefined]} isLastSorting={false} onFilter={[undefined]} currFilters={[undefined]} onExternalFilter={[undefined]} filterPosition=\\"inline\\">
                                                    <th tabIndex={0} aria-label=\\"Location sortable\\" onKeyUp={[Function]} onClick={[Function]} className=\\"sortable\\">
                                                      Location
                                                      <SortSymbol>
                                                        <span className=\\"order-4\\" />
                                                      </SortSymbol>
                                                    </th>
                                                  </HeaderCell>
                                                  <HeaderCell index={2} column={{...}} onSort={[Function]} sorting={false} sortOrder=\\"asc\\" globalSortCaret={[undefined]} isLastSorting={false} onFilter={[undefined]} currFilters={[undefined]} onExternalFilter={[undefined]} filterPosition=\\"inline\\">
                                                    <th tabIndex={0} aria-label=\\"Last SignIn Date sortable\\" onKeyUp={[Function]} onClick={[Function]} className=\\"sortable\\">
                                                      Last SignIn Date
                                                      <SortSymbol>
                                                        <span className=\\"order-4\\" />
                                                      </SortSymbol>
                                                    </th>
                                                  </HeaderCell>
                                                  <HeaderCell index={3} column={{...}} onSort={[Function]} sorting={false} sortOrder=\\"asc\\" globalSortCaret={[undefined]} isLastSorting={false} onFilter={[undefined]} currFilters={[undefined]} onExternalFilter={[undefined]} filterPosition=\\"inline\\">
                                                    <th tabIndex={0} className=\\"column-right\\">
                                                      Results: 2
                                                    </th>
                                                  </HeaderCell>
                                                </tr>
                                              </thead>
                                            </Header>
                                            <Body className={[undefined]} data={{...}} keyField=\\"id\\" tabIndexCell={[undefined]} columns={{...}} isEmpty={false} visibleColumnSize={4} noDataIndication=\\"No Data\\" cellEdit={{...}} selectRow={{...}} expandRow={{...}} rowStyle={[undefined]} rowClasses={[undefined]} rowEvents={[undefined]}>
                                              <tbody className={[undefined]}>
                                                <SimpleRow row={{...}} tabIndexCell={[undefined]} columns={{...}} keyField=\\"id\\" cellEdit={{...}} value=\\"1\\" rowIndex={0} visibleColumnSize={4} attrs={{...}} style={{...}} className={{...}} editable={true}>
                                                  <tr style={{...}} className={{...}}>
                                                    <RowPureContent shouldUpdate={true} tabIndexStart={-1} row={{...}} columns={{...}} keyField=\\"id\\" cellEdit={{...}} value=\\"1\\" rowIndex={0} editable={true}>
                                                      <Cell row={{...}} editable={true} rowIndex={0} columnIndex={0} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]}>
                                                        <td>
                                                          testuser@kujakuja.com
                                                        </td>
                                                      </Cell>
                                                      <Cell row={{...}} editable={true} rowIndex={0} columnIndex={1} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]}>
                                                        <td>
                                                          Test Location
                                                        </td>
                                                      </Cell>
                                                      <Cell row={{...}} editable={true} rowIndex={0} columnIndex={2} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]}>
                                                        <td />
                                                      </Cell>
                                                      <Cell row={{...}} editable={true} rowIndex={0} columnIndex={3} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]} className=\\"column-right\\">
                                                        <td className=\\"column-right\\">
                                                          <Button color=\\"primary\\" className=\\"primary\\" onClick={[Function: onClick]} tag=\\"button\\">
                                                            <button type=\\"button\\" onClick={[Function: bound onClick]} className=\\"primary btn btn-primary\\" aria-label={{...}}>
                                                              View
                                                            </button>
                                                          </Button>
                                                        </td>
                                                      </Cell>
                                                    </RowPureContent>
                                                  </tr>
                                                </SimpleRow>
                                                <SimpleRow row={{...}} tabIndexCell={[undefined]} columns={{...}} keyField=\\"id\\" cellEdit={{...}} value=\\"2\\" rowIndex={1} visibleColumnSize={4} attrs={{...}} style={{...}} className={{...}} editable={true}>
                                                  <tr style={{...}} className={{...}}>
                                                    <RowPureContent shouldUpdate={true} tabIndexStart={-1} row={{...}} columns={{...}} keyField=\\"id\\" cellEdit={{...}} value=\\"2\\" rowIndex={1} editable={true}>
                                                      <Cell row={{...}} editable={true} rowIndex={1} columnIndex={0} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]}>
                                                        <td>
                                                          testuser2@kujakuja.com
                                                        </td>
                                                      </Cell>
                                                      <Cell row={{...}} editable={true} rowIndex={1} columnIndex={1} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]}>
                                                        <td>
                                                          Test Location
                                                        </td>
                                                      </Cell>
                                                      <Cell row={{...}} editable={true} rowIndex={1} columnIndex={2} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]}>
                                                        <td />
                                                      </Cell>
                                                      <Cell row={{...}} editable={true} rowIndex={1} columnIndex={3} column={{...}} onStart={[undefined]} clickToEdit={[undefined]} dbclickToEdit={[undefined]} className=\\"column-right\\">
                                                        <td className=\\"column-right\\">
                                                          <Button color=\\"primary\\" className=\\"primary\\" onClick={[Function: onClick]} tag=\\"button\\">
                                                            <button type=\\"button\\" onClick={[Function: bound onClick]} className=\\"primary btn btn-primary\\" aria-label={{...}}>
                                                              View
                                                            </button>
                                                          </Button>
                                                        </td>
                                                      </Cell>
                                                    </RowPureContent>
                                                  </tr>
                                                </SimpleRow>
                                              </tbody>
                                            </Body>
                                          </table>
                                        </div>
                                      </BootstrapTable>
                                    </ColumnManagementProvider>
                                  </PaginationDataProvider>
                                </SortProvider>
                              </DataProvider>
                            </BootstrapTableContainer>
                            <Component custom={true} page={1} totalSize={2} sizePerPage={25} sizePerPageList={{...}} bootstrap4={[undefined]} pageStartIndex={1} hidePageListOnlyOnePage={false} hideSizePerPage={false} alwaysShowAllBtns={false} withFirstAndLast={true} dataSize={2} paginationSize={5} showTotal={[undefined]} pageListRenderer={[undefined]} pageButtonRenderer={[undefined]} sizePerPageRenderer={[undefined]} paginationTotalRenderer={[undefined]} sizePerPageOptionRenderer={[undefined]} firstPageText=\\"<<\\" prePageText=\\"<\\" nextPageText=\\">\\" lastPageText=\\">>\\" prePageTitle=\\"previous page\\" nextPageTitle=\\"next page\\" firstPageTitle=\\"first page\\" lastPageTitle=\\"last page\\" onPageChange={[Function: bound handleChangePage]} onSizePerPageChange={[Function: bound handleChangeSizePerPage]}>
                              <PaginationHandler custom={true} totalSize={2} sizePerPageList={{...}} bootstrap4={[undefined]} pageStartIndex={1} hidePageListOnlyOnePage={false} hideSizePerPage={false} alwaysShowAllBtns={false} withFirstAndLast={true} dataSize={2} paginationSize={5} showTotal={[undefined]} pageListRenderer={[undefined]} pageButtonRenderer={[undefined]} sizePerPageRenderer={[undefined]} paginationTotalRenderer={[undefined]} sizePerPageOptionRenderer={[undefined]} firstPageText=\\"<<\\" prePageText=\\"<\\" nextPageText=\\">\\" lastPageText=\\">>\\" prePageTitle=\\"previous page\\" nextPageTitle=\\"next page\\" firstPageTitle=\\"first page\\" lastPageTitle=\\"last page\\" onPageChange={[Function: bound handleChangePage]} onSizePerPageChange={[Function: bound handleChangeSizePerPage]} currPage={1} currSizePerPage={25}>
                                <SizePerPageDropdownAdapter custom={true} totalSize={2} sizePerPageList={{...}} bootstrap4={[undefined]} pageStartIndex={1} hidePageListOnlyOnePage={false} hideSizePerPage={false} alwaysShowAllBtns={false} withFirstAndLast={true} dataSize={2} paginationSize={5} showTotal={[undefined]} pageListRenderer={[undefined]} pageButtonRenderer={[undefined]} sizePerPageRenderer={[undefined]} paginationTotalRenderer={[undefined]} sizePerPageOptionRenderer={[undefined]} firstPageText=\\"<<\\" prePageText=\\"<\\" nextPageText=\\">\\" lastPageText=\\">>\\" prePageTitle=\\"previous page\\" nextPageTitle=\\"next page\\" firstPageTitle=\\"first page\\" lastPageTitle=\\"last page\\" onPageChange={[Function: bound handleChangePage]} onSizePerPageChange={[Function: bound handleChangeSizePerPage]} currPage={1} currSizePerPage={25} lastPage={1} totalPages={1}>
                                  <SizePerPageDropdownStandalone custom={true} totalSize={2} sizePerPageList={{...}} bootstrap4={[undefined]} pageStartIndex={1} hidePageListOnlyOnePage={false} hideSizePerPage={false} alwaysShowAllBtns={false} withFirstAndLast={true} dataSize={2} paginationSize={5} showTotal={[undefined]} pageListRenderer={[undefined]} pageButtonRenderer={[undefined]} sizePerPageRenderer={[undefined]} paginationTotalRenderer={[undefined]} sizePerPageOptionRenderer={[undefined]} firstPageText=\\"<<\\" prePageText=\\"<\\" nextPageText=\\">\\" lastPageText=\\">>\\" prePageTitle=\\"previous page\\" nextPageTitle=\\"next page\\" firstPageTitle=\\"first page\\" lastPageTitle=\\"last page\\" onPageChange={[Function: bound handleChangePage]} onSizePerPageChange={[Function: bound handleChangeSizePerPage]} currPage={1} currSizePerPage=\\"25\\" lastPage={1} totalPages={1} options={{...}} optionRenderer={[undefined]} onClick={[Function: bound toggleDropDown]} onBlur={[Function: bound closeDropDown]} open={false}>
                                    <SizePerPageDropDown custom={true} totalSize={2} sizePerPageList={{...}} bootstrap4={[undefined]} pageStartIndex={1} hidePageListOnlyOnePage={false} hideSizePerPage={false} alwaysShowAllBtns={false} withFirstAndLast={true} dataSize={2} paginationSize={5} showTotal={[undefined]} pageListRenderer={[undefined]} pageButtonRenderer={[undefined]} sizePerPageRenderer={[undefined]} paginationTotalRenderer={[undefined]} sizePerPageOptionRenderer={[undefined]} firstPageText=\\"<<\\" prePageText=\\"<\\" nextPageText=\\">\\" lastPageText=\\">>\\" prePageTitle=\\"previous page\\" nextPageTitle=\\"next page\\" firstPageTitle=\\"first page\\" lastPageTitle=\\"last page\\" onPageChange={[Function: bound handleChangePage]} onSizePerPageChange={[Function: bound handleChangeSizePerPage]} currPage={1} currSizePerPage=\\"25\\" lastPage={1} totalPages={1} options={{...}} optionRenderer={{...}} onClick={[Function: bound toggleDropDown]} onBlur={[Function: bound closeDropDown]} open={false} hidden={false} btnContextual=\\"btn-default btn-secondary\\" variation=\\"dropdown\\" className=\\"\\">
                                      <span style={{...}} className=\\"react-bs-table-sizePerPage-dropdown dropdown\\">
                                        <button id=\\"pageDropDown\\" className=\\"btn btn-default btn-secondary dropdown-toggle\\" data-toggle=\\"dropdown\\" aria-expanded={false} onClick={[Function: bound toggleDropDown]} onBlur={[Function: bound closeDropDown]}>
                                          25
                                           
                                          <span>
                                            <span className=\\"caret\\" />
                                          </span>
                                        </button>
                                        <ul className=\\"dropdown-menu \\" role=\\"menu\\" aria-labelledby=\\"pageDropDown\\">
                                          <SizePerPageOption text=\\"25\\" page={25} bootstrap4={false} onSizePerPageChange={[Function: bound handleChangeSizePerPage]}>
                                            <li role=\\"presentation\\" className=\\"dropdown-item\\">
                                              <a href=\\"#\\" tabIndex=\\"-1\\" role=\\"menuitem\\" data-page={25} onMouseDown={[Function: onMouseDown]}>
                                                25
                                              </a>
                                            </li>
                                          </SizePerPageOption>
                                          <SizePerPageOption text=\\"50\\" page={50} bootstrap4={false} onSizePerPageChange={[Function: bound handleChangeSizePerPage]}>
                                            <li role=\\"presentation\\" className=\\"dropdown-item\\">
                                              <a href=\\"#\\" tabIndex=\\"-1\\" role=\\"menuitem\\" data-page={50} onMouseDown={[Function: onMouseDown]}>
                                                50
                                              </a>
                                            </li>
                                          </SizePerPageOption>
                                          <SizePerPageOption text=\\"100\\" page={100} bootstrap4={false} onSizePerPageChange={[Function: bound handleChangeSizePerPage]}>
                                            <li role=\\"presentation\\" className=\\"dropdown-item\\">
                                              <a href=\\"#\\" tabIndex=\\"-1\\" role=\\"menuitem\\" data-page={100} onMouseDown={[Function: onMouseDown]}>
                                                100
                                              </a>
                                            </li>
                                          </SizePerPageOption>
                                          <SizePerPageOption text=\\"ALL\\" page={2} bootstrap4={false} onSizePerPageChange={[Function: bound handleChangeSizePerPage]}>
                                            <li role=\\"presentation\\" className=\\"dropdown-item\\">
                                              <a href=\\"#\\" tabIndex=\\"-1\\" role=\\"menuitem\\" data-page={2} onMouseDown={[Function: onMouseDown]}>
                                                ALL
                                              </a>
                                            </li>
                                          </SizePerPageOption>
                                        </ul>
                                      </span>
                                    </SizePerPageDropDown>
                                  </SizePerPageDropdownStandalone>
                                </SizePerPageDropdownAdapter>
                              </PaginationHandler>
                            </Component>
                            <Component custom={true} page={1} totalSize={2} sizePerPage={25} sizePerPageList={{...}} bootstrap4={[undefined]} pageStartIndex={1} hidePageListOnlyOnePage={false} hideSizePerPage={false} alwaysShowAllBtns={false} withFirstAndLast={true} dataSize={2} paginationSize={5} showTotal={[undefined]} pageListRenderer={[undefined]} pageButtonRenderer={[undefined]} sizePerPageRenderer={[undefined]} paginationTotalRenderer={[undefined]} sizePerPageOptionRenderer={[undefined]} firstPageText=\\"<<\\" prePageText=\\"<\\" nextPageText=\\">\\" lastPageText=\\">>\\" prePageTitle=\\"previous page\\" nextPageTitle=\\"next page\\" firstPageTitle=\\"first page\\" lastPageTitle=\\"last page\\" onPageChange={[Function: bound handleChangePage]} onSizePerPageChange={[Function: bound handleChangeSizePerPage]}>
                              <PaginationHandler custom={true} totalSize={2} sizePerPageList={{...}} bootstrap4={[undefined]} pageStartIndex={1} hidePageListOnlyOnePage={false} hideSizePerPage={false} alwaysShowAllBtns={false} withFirstAndLast={true} dataSize={2} paginationSize={5} showTotal={[undefined]} pageListRenderer={[undefined]} pageButtonRenderer={[undefined]} sizePerPageRenderer={[undefined]} paginationTotalRenderer={[undefined]} sizePerPageOptionRenderer={[undefined]} firstPageText=\\"<<\\" prePageText=\\"<\\" nextPageText=\\">\\" lastPageText=\\">>\\" prePageTitle=\\"previous page\\" nextPageTitle=\\"next page\\" firstPageTitle=\\"first page\\" lastPageTitle=\\"last page\\" onPageChange={[Function: bound handleChangePage]} onSizePerPageChange={[Function: bound handleChangeSizePerPage]} currPage={1} currSizePerPage={25}>
                                <PaginationListAdapter custom={true} totalSize={2} sizePerPageList={{...}} bootstrap4={[undefined]} pageStartIndex={1} hidePageListOnlyOnePage={false} hideSizePerPage={false} alwaysShowAllBtns={false} withFirstAndLast={true} dataSize={2} paginationSize={5} showTotal={[undefined]} pageListRenderer={[undefined]} pageButtonRenderer={[undefined]} sizePerPageRenderer={[undefined]} paginationTotalRenderer={[undefined]} sizePerPageOptionRenderer={[undefined]} firstPageText=\\"<<\\" prePageText=\\"<\\" nextPageText=\\">\\" lastPageText=\\">>\\" prePageTitle=\\"previous page\\" nextPageTitle=\\"next page\\" firstPageTitle=\\"first page\\" lastPageTitle=\\"last page\\" onPageChange={[Function: bound handleChangePage]} onSizePerPageChange={[Function: bound handleChangeSizePerPage]} currPage={1} currSizePerPage={25} lastPage={1} totalPages={1}>
                                  <PaginationListStandalone pageButtonRenderer={[undefined]} onPageChange={[Function: bound handleChangePage]} pages={{...}}>
                                    <PaginatonList pageButtonRenderer={{...}} onPageChange={[Function: bound handleChangePage]} pages={{...}}>
                                      <ul className=\\"pagination react-bootstrap-table-page-btns-ul\\">
                                        <PageButton page={1} active={true} disabled={false} title=\\"1\\" onPageChange={[Function: bound handleChangePage]}>
                                          <li className=\\"active page-item\\" title=\\"1\\">
                                            <a href=\\"#\\" onClick={[Function: bound handleClick]} className=\\"page-link\\">
                                              1
                                            </a>
                                          </li>
                                        </PageButton>
                                      </ul>
                                    </PaginatonList>
                                  </PaginationListStandalone>
                                </PaginationListAdapter>
                              </PaginationHandler>
                            </Component>
                          </div>
                        </StateProvider>
                      </CustomizableProvider>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </div>
        </Container>
      </div>
    </ToolkitProvider>
  </Toolkitprovider>
</UsersList>"
`;
